Userland/DevTools/UserspaceEmulator/SoftFPU.cpp

```diff
@@ -1024,9 +1024,16 @@ void SoftFPU::F2XM1(const X86::Instruction&)
 {
     // FIXME: Raise #IA #D #U #P
     // FIXME: Set C1 on when result was rounded up, cleared otherwise
-    // FIXME: Validate ST(0) is in range –1.0 to +1.0
     auto val = fpu_get(0);
-    fpu_set(0, exp2(val) - 1.0l);
+    // Validate ST(0) is in range –1.0 to +1.0.
+    // If the source value is outside this range, the result is undefined.
+    // In practice, the result is the input value.
+    if (val >= -1.0 && val <= 1.0) {
+        fpu_set(0, exp2(val) - 1.0l);
+    } else {
+        fpu_set(0, val);
+        reportln("\033[31;1mWarning! Source operand for F2XM1 is out of range! ({}, should be in range -1.0 to +1.0)\033[0m\n"sv, val);
```

Comment:
I think c1 should be cleared in this path,
Can you also check if this path sets any exception flag, #IA seems like it might fit.

Reply:
`C1` (Overflow flag) should definitely be set/cleared in both paths, or at least the main path, there is a FIXME for this :^)

`#IA` is only reported for signalling NaN (or when the input point format is unsupported, but I can't test that right now). I do not get `#IA` for any other value. I do get `FE_INEXACT` (`#U`), but again, there is a FIXME for that.

I might look into those FIXME's tomorrow, but actually this was just a FIXME roulette :^)